File: rasterizer-dotnet-vb/RasterizationEventHandling.vb
Imports ceTe.DynamicPDF.Rasterizer
Imports System

Class RasterizationEventHandling

    Public Shared Sub Run()
        PngEventHandling()
        TiffEventHandling()
    End Sub

    Public Shared Sub PngEventHandling()
        Console.WriteLine("----- processing PNG -------")
        Dim rasterizer As New PdfRasterizer(Program.GetResourcePath("DocumentA.pdf"))
        AddHandler rasterizer.RasterizerProgress, AddressOf RasterizerProgressing

        Dim fixedImageSize As New FixedImageSize(595, 841)
        Dim oPngImageFormat As New PngImageFormat(PngColorFormat.RgbA)
        rasterizer.Draw(Program.GetOutputDocPath("eventHandlingExample.png"), oPngImageFormat, fixedImageSize)
        rasterizer.Dispose()
    End Sub

    Friend Shared Sub TiffEventHandling()
        Console.WriteLine("------- processing MultiPage Tiff ------")
        Dim rasterizer As New PdfRasterizer(Program.GetResourcePath("DocumentA.pdf"))
        AddHandler rasterizer.RasterizerProgress, AddressOf RasterizerProgressing

        Dim fixedImageSize As New FixedImageSize(595, 841)
        rasterizer.DrawToMultiPageTiff(Program.GetOutputDocPath("eventHandlingMultiPageTiff.tif"), ImageFormat.TiffWithLzw, fixedImageSize)
    End Sub

    Private Shared Sub RasterizerProgressing(sender As Object, e As RasterizerProgressChangedEventArgs)
        Console.WriteLine($"{e.PagesCompleted} pages have been rasterized")
    End Sub

End Class

File: rasterizer-dotnet-examples-csharp/RasterizationEventHandling.cs
using ceTe.DynamicPDF.Rasterizer;
using System;


namespace rasterizer_dotnet_csharp
{
    class RasterizationEventHandling
    {
        public static void Run()
        {
            PngEventHandling();
            TiffEventHandling();
        }
        
        public static void PngEventHandling()
        {
            Console.WriteLine("----- processing PNG -------");
            PdfRasterizer rasterizer = new PdfRasterizer(Program.GetResourcePath("DocumentA.pdf"));
            rasterizer.RasterizerProgress += new ProgressChangedEventHandler(RasterizerProgressing);
            FixedImageSize fixedImageSize = new FixedImageSize(595, 841);
            PngImageFormat oPngImageFormat = new PngImageFormat(PngColorFormat.RgbA);
            rasterizer.Draw(Program.GetOutputDocPath("eventHandlingExample.png"), oPngImageFormat, fixedImageSize);
            rasterizer.Dispose();
        }

        internal static void TiffEventHandling()
        {
            Console.WriteLine("------- processing MultiPage Tiff ------");
            PdfRasterizer rasterizer = new PdfRasterizer(Program.GetResourcePath("DocumentA.pdf"));
            rasterizer.RasterizerProgress += new ProgressChangedEventHandler(RasterizerProgressing);

            // Create a image size tha is a fixed size
            FixedImageSize fixedImageSize = new FixedImageSize(595, 841);

            // Save the image
            rasterizer.DrawToMultiPageTiff(Program.GetOutputDocPath("eventHandlingMultiPageTiff.tif"), ImageFormat.TiffWithLzw, fixedImageSize);
        }

        static void RasterizerProgressing(object sender, RasterizerProgressChangedEventArgs e)
        {
            Console.WriteLine(e.PagesCompleted + " pages have been rasterized");
        }

    }
}
